GitCommands/Settings/ConfigFileSettings.cs

```diff
@@ -70,16 +70,26 @@ public static ConfigFileSettings CreateGlobal(ConfigFileSettings? lowerPriority,
                 SettingLevel.SystemWide);
         }
 
-        public new string GetValue(string setting)
-        {
-            return GetString(setting, string.Empty);
-        }
+        public new string GetValue(string setting) => GetString(setting, string.Empty);
+
+        public T? GetValue<T>(string setting) where T : struct => ConvertValue<T>(GetValue(setting), setting);
 
-        public IReadOnlyList<string> GetValues(string setting)
+        private T? ConvertValue<T>(string value, string setting) where T : struct
         {
-            return SettingsCache.GetValues(setting);
+            Type targetType = typeof(T);
+            try
+            {
+                return (T)Convert.ChangeType(value, targetType);
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine($"Setting '{setting}': fail to convert value '{value}' into type '{targetType}'. Error: {ex}");
+                return null;
```

Comment:
```suggestion
                return default(T);
```

Reply:
Why handle the exception at all?
If we try to get a settings using a wrong type, it is a bug.

Reply:
I agree

Reply:
It was at least for when the setting was not set. But indeed it is better to handle it before. Done.